Tests failing

I'm getting the following test failures:
~~~~~
+ py.test-2.7 -v jupyter_core/tests/
============================= test session starts ==============================
platform linux2 -- Python 2.7.10 -- py-1.4.29 -- pytest-2.6.4 -- /usr/bin/python
collecting ... collected 38 items

jupyter_core/tests/test_application.py::test_basic PASSED
jupyter_core/tests/test_application.py::test_default_traits PASSED
jupyter_core/tests/test_application.py::test_custom_config PASSED
jupyter_core/tests/test_application.py::test_cli_override PASSED
jupyter_core/tests/test_application.py::test_generate_config PASSED
jupyter_core/tests/test_application.py::test_load_config FAILED
jupyter_core/tests/test_command.py::test_config_dir FAILED
jupyter_core/tests/test_command.py::test_data_dir FAILED
jupyter_core/tests/test_command.py::test_runtime_dir FAILED
jupyter_core/tests/test_command.py::test_paths FAILED
jupyter_core/tests/test_command.py::test_paths_json FAILED
jupyter_core/tests/test_command.py::test_subcommand_not_given FAILED
jupyter_core/tests/test_command.py::test_help FAILED
jupyter_core/tests/test_command.py::test_subcommand_not_found FAILED
jupyter_core/tests/test_command.py::test_subcommand_list PASSED
jupyter_core/tests/test_migrate.py::test_migrate_file PASSED
jupyter_core/tests/test_migrate.py::test_migrate_dir PASSED
jupyter_core/tests/test_migrate.py::test_migrate_one PASSED
jupyter_core/tests/test_migrate.py::test_migrate_config PASSED
jupyter_core/tests/test_migrate.py::test_migrate_custom_default PASSED
jupyter_core/tests/test_migrate.py::test_migrate_nothing PASSED
jupyter_core/tests/test_migrate.py::test_migrate_default PASSED
jupyter_core/tests/test_migrate.py::test_migrate PASSED
jupyter_core/tests/test_paths.py::test_config_dir_darwin PASSED
jupyter_core/tests/test_paths.py::test_config_dir_windows PASSED
jupyter_core/tests/test_paths.py::test_config_dir_linux PASSED
jupyter_core/tests/test_paths.py::test_data_dir_env PASSED
jupyter_core/tests/test_paths.py::test_data_dir_darwin PASSED
jupyter_core/tests/test_paths.py::test_data_dir_windows PASSED
jupyter_core/tests/test_paths.py::test_data_dir_linux PASSED
jupyter_core/tests/test_paths.py::test_runtime_dir_env PASSED
jupyter_core/tests/test_paths.py::test_runtime_dir_darwin PASSED
jupyter_core/tests/test_paths.py::test_runtime_dir_windows PASSED
jupyter_core/tests/test_paths.py::test_runtime_dir_linux PASSED
jupyter_core/tests/test_paths.py::test_jupyter_path PASSED
jupyter_core/tests/test_paths.py::test_jupyter_path_env PASSED
jupyter_core/tests/test_paths.py::test_jupyter_path_sys_prefix PASSED
jupyter_core/tests/test_paths.py::test_jupyter_path_subdir PASSED

=================================== FAILURES ===================================
_______________________________ test_load_config _______________________________

    def test_load_config():
        config_dir = mkdtemp()
        wd = mkdtemp()
        with open(pjoin(config_dir, 'dummy_app_config.py'), 'w') as f:
            f.write('c.DummyApp.m = 1\n')
            f.write('c.DummyApp.n = 1')
        with patch.object(py3compat, 'getcwd', lambda : wd):
            app = DummyApp(config_dir=config_dir)
>           app.initialize()

jupyter_core/tests/test_application.py:80: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
<string>:2: in initialize
    ???
/usr/lib/python2.7/site-packages/traitlets/config/application.py:75: in catch_config_error
    return method(app, *args, **kwargs)
jupyter_core/application.py:233: in initialize
    self.parse_command_line(argv)
<string>:2: in parse_command_line
    ???
/usr/lib/python2.7/site-packages/traitlets/config/application.py:81: in catch_config_error
    app.exit(1)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <jupyter_core.tests.test_application.DummyApp object at 0x7f2fa7997410>
exit_status = 1

    def exit(self, exit_status=0):
        self.log.debug("Exiting application: %s" % self.name)
>       sys.exit(exit_status)
E       SystemExit: 1

/usr/lib/python2.7/site-packages/traitlets/config/application.py:582: SystemExit
----------------------------- Captured stdout call -----------------------------
A Jupyter Application

Options
-------

Arguments that take values are actually convenience aliases to full
Configurables, whose aliases are listed on the help line. For more information
on full configurables, see '--help-all'.

--debug
    set log level to logging.DEBUG (maximize logging output)
-y
    Answer yes to any questions instead of prompting.
--generate-config
    generate default config file
--config=<Unicode> (JupyterApp.config_file)
    Default: u''
    Full path of a config file.
--log-level=<Enum> (Application.log_level)
    Default: 30
    Choices: (0, 10, 20, 30, 40, 50, 'DEBUG', 'INFO', 'WARN', 'ERROR', 'CRITICAL')
    Set the log level by value or name.

To see all available configurables, use `--help-all`

----------------------------- Captured stderr call -----------------------------
[DummyApp] CRITICAL | Bad config encountered during initialization:
[DummyApp] CRITICAL | Unrecognized flag: '-v'
_______________________________ test_config_dir ________________________________

    def test_config_dir():
>       assert_output('--config-dir', jupyter_config_dir())

jupyter_core/tests/test_command.py:35: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
jupyter_core/tests/test_command.py:31: in assert_output
    assert get_jupyter_output(cmd) == expected
jupyter_core/tests/test_command.py:27: in get_jupyter_output
    return check_output(['jupyter'] + cmd).decode('utf8').strip()
/usr/lib64/python2.7/subprocess.py:566: in check_output
    process = Popen(stdout=PIPE, *popenargs, **kwargs)
/usr/lib64/python2.7/subprocess.py:710: in __init__
    errread, errwrite)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
~~~~

Thanks, still have other failures.  First is:
~~~~~
=================================== FAILURES ===================================
_______________________________ test_config_dir ________________________________

    def test_config_dir():
>       assert_output('--config-dir', jupyter_config_dir())

jupyter_core/tests/test_command.py:35: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
jupyter_core/tests/test_command.py:31: in assert_output
    assert get_jupyter_output(cmd) == expected
jupyter_core/tests/test_command.py:27: in get_jupyter_output
    return check_output(['jupyter'] + cmd).decode('utf8').strip()
/usr/lib64/python2.7/subprocess.py:566: in check_output
    process = Popen(stdout=PIPE, *popenargs, **kwargs)
/usr/lib64/python2.7/subprocess.py:710: in __init__
    errread, errwrite)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <subprocess.Popen object at 0x7feb082419d0>
args = ['jupyter', '--config-dir'], executable = 'jupyter', preexec_fn = None
close_fds = False, cwd = None, env = None, universal_newlines = False
startupinfo = None, creationflags = 0, shell = False, to_close = set([8])
p2cread = None, p2cwrite = None, c2pread = 8, c2pwrite = 10, errread = None
errwrite = None

    def _execute_child(self, args, executable, preexec_fn, close_fds,
                       cwd, env, universal_newlines,
                       startupinfo, creationflags, shell, to_close,
                       p2cread, p2cwrite,
                       c2pread, c2pwrite,
                       errread, errwrite):
        """Execute program (POSIX version)"""
    
        if isinstance(args, types.StringTypes):
            args = [args]
        else:
            args = list(args)
    
        if shell:
            args = ["/bin/sh", "-c"] + args
            if executable:
                args[0] = executable
    
        if executable is None:
            executable = args[0]
    
        def _close_in_parent(fd):
            os.close(fd)
            to_close.remove(fd)
    
        # For transferring possible exec failure from child to parent
        # The first char specifies the exception type: 0 means
        # OSError, 1 means some other error.
        errpipe_read, errpipe_write = self.pipe_cloexec()
        try:
            try:
                gc_was_enabled = gc.isenabled()
                # Disable gc to avoid bug where gc -> file_dealloc ->
                # write to stderr -> hang.  http://bugs.python.org/issue1336
                gc.disable()
                try:
                    self.pid = os.fork()
                except:
                    if gc_was_enabled:
                        gc.enable()
                    raise
                self._child_created = True
                if self.pid == 0:
                    # Child
                    try:
                        # Close parent's pipe ends
                        if p2cwrite is not None:
                            os.close(p2cwrite)
                        if c2pread is not None:
                            os.close(c2pread)
                        if errread is not None:
                            os.close(errread)
                        os.close(errpipe_read)
    
                        # When duping fds, if there arises a situation
                        # where one of the fds is either 0, 1 or 2, it
                        # is possible that it is overwritten (#12607).
                        if c2pwrite == 0:
                            c2pwrite = os.dup(c2pwrite)
                        if errwrite == 0 or errwrite == 1:
                            errwrite = os.dup(errwrite)
    
                        # Dup fds for child
                        def _dup2(a, b):
                            # dup2() removes the CLOEXEC flag but
                            # we must do it ourselves if dup2()
                            # would be a no-op (issue #10806).
                            if a == b:
                                self._set_cloexec_flag(a, False)
                            elif a is not None:
                                os.dup2(a, b)
                        _dup2(p2cread, 0)
                        _dup2(c2pwrite, 1)
                        _dup2(errwrite, 2)
    
                        # Close pipe fds.  Make sure we don't close the
                        # same fd more than once, or standard fds.
                        closed = { None }
                        for fd in [p2cread, c2pwrite, errwrite]:
                            if fd not in closed and fd > 2:
                                os.close(fd)
                                closed.add(fd)
    
                        if cwd is not None:
                            os.chdir(cwd)
    
                        if preexec_fn:
                            preexec_fn()
    
                        # Close all other fds, if asked for - after
                        # preexec_fn(), which may open FDs.
                        if close_fds:
                            self._close_fds(but=errpipe_write)
    
                        if env is None:
                            os.execvp(executable, args)
                        else:
                            os.execvpe(executable, args, env)
    
                    except:
                        exc_type, exc_value, tb = sys.exc_info()
                        # Save the traceback and attach it to the exception object
                        exc_lines = traceback.format_exception(exc_type,
                                                               exc_value,
                                                               tb)
                        exc_value.child_traceback = ''.join(exc_lines)
                        os.write(errpipe_write, pickle.dumps(exc_value))
    
                    # This exitcode won't be reported to applications, so it
                    # really doesn't matter what we return.
                    os._exit(255)
    
                # Parent
                if gc_was_enabled:
                    gc.enable()
            finally:
                # be sure the FD is closed no matter what
                os.close(errpipe_write)
    
            # Wait for exec to fail or succeed; possibly raising exception
            # Exception limited to 1M
            data = _eintr_retry_call(os.read, errpipe_read, 1048576)
        finally:
            if p2cread is not None and p2cwrite is not None:
                _close_in_parent(p2cread)
            if c2pwrite is not None and c2pread is not None:
                _close_in_parent(c2pwrite)
            if errwrite is not None and errread is not None:
                _close_in_parent(errwrite)
    
            # be sure the FD is closed no matter what
            os.close(errpipe_read)
    
        if data != "":
            try:
                _eintr_retry_call(os.waitpid, self.pid, 0)
            except OSError as e:
                if e.errno != errno.ECHILD:
                    raise
            child_exception = pickle.loads(data)
>           raise child_exception
E           OSError: [Errno 2] No such file or directory

/usr/lib64/python2.7/subprocess.py:1335: OSError
~~~~~~
Am I missing test data?

--- a/jupyter_core/tests/test_application.py
+++ b/jupyter_core/tests/test_application.py
@@ -77,7 +77,7 @@ def test_load_config():
         f.write('c.DummyApp.n = 1')
     with patch.object(py3compat, 'getcwd', lambda : wd):
         app = DummyApp(config_dir=config_dir)
-        app.initialize()
+        app.initialize([])
 
     assert app.n == 1, "Loaded config from config dir"
     
@@ -86,7 +86,7 @@ def test_load_config():
 
     with patch.object(py3compat, 'getcwd', lambda : wd):
         app = DummyApp(config_dir=config_dir)
-        app.initialize()
+        app.initialize([])
 
     assert app.m == 1, "Loaded config from config dir"
     assert app.n == 2, "Loaded config from CWD"

